src/helpers/ht_device_info.service.ts

```diff
@@ -70,26 +50,344 @@ class DeviceInfoService {
+                const values = this.fingerprintComponents.map(function (component) { return component.value });
+                this.fingerprint = Fingerprint2.x64hash128(values.join(''), 31);
+                this.pushFPC(this.fingerprintDetails, 'fingerprint', this.fingerprint);
+            } else {
+                return null;
+            }
+        }
+        return this.fingerprint;
+    }
+
+    getFingerprintComponents() {
+
+        if ( !this.fingerprintDetails ) {
+            this.fingerprintDetails = new Array();
+
+            if ( this.fingerprintComponents ) {
+                for ( let i = 0; i < this.fingerprintComponents.length; i++ ) {
+                    if ( this.fingerprintComponents[i] &&
+                        this.fingerprintComponents[i].key !== 'canvas' &&
+                        this.fingerprintComponents[i].key !== 'webgl' ) {
+
+                        if ( this.fingerprintComponents[i].key === 'js_fonts' ) {
+                            this.fonts = this.fingerprintComponents[i].key;
+                        }
+                        this.fingerprintDetails.push({
+                            key: this.fingerprintComponents[i].key,
+                            value: this.joinValues(this.fingerprintComponents[i].value)
+                        });
+                    }
+                }
+
+
+                // FingerprintJS does not generate these
+                this.pushFPC(this.fingerprintDetails, 'osType', this.getOsType());
+                this.pushFPC(this.fingerprintDetails, 'browserType', this.getBrowser());
+                this.pushFPC(this.fingerprintDetails, 'browserVersion', this.getBrowserVersion());
+                this.pushFPC(this.fingerprintDetails, 'deviceName', this.getDeviceName());
+                this.pushFPC(this.fingerprintDetails, 'userAgent', navigator.userAgent);
+
+                this.getFingerprint();
+
+                if ( this.getBrowserVersion() ) {
+                    this.pushFPC(this.fingerprintDetails, 'browser_major_version', this.getBrowserMajorVersion());
+                }
+
+                if ( this.browserData ) {
+                    if ( this.browserData.engine ) {
+                        this.pushFPC(this.fingerprintDetails, 'engine', this.browserData.engine.name);
+                        this.pushFPC(this.fingerprintDetails, 'engine_version', this.browserData.engine.version);
+                    }
+                    if ( this.browserData.os ) {
+                        this.pushFPC(this.fingerprintDetails, 'engine_version', this.browserData.os.version);
+                    }
+                    if ( this.browserData.device ) {
+                        this.pushFPC(this.fingerprintDetails, 'device_type', this.browserData.device.type);
+                        this.pushFPC(this.fingerprintDetails, 'device_vendor', this.browserData.device.vendor);
+                    }
+                    if ( this.browserData.cpu ) {
```

Comment:
Is the cpu or the architecture the one you would like to add? The if check one and then you set the other one.

Reply:
Fixed, should be browserData.cpu.architecture, good catch thanks